djangocms_versioning_filer/monkeypatch/admin/fileadmin.py

```diff
@@ -9,7 +10,11 @@ def save_model(func):
     """Override the FileAdmin save_model method"""
     def inner(self, request, obj, form, change):
         func(self, request, obj, form, change)
-
+        if isinstance(obj, File):
```

Comment:
This is better.

@NarenderRajuB I don't think that we need to check if this is a File object here because we do a check below if the object has an id and a grouper attached. If the model was say Grouper then it wouldn't have obj.grouper.

If you added this because an error occurred we may have to run a check that the object has the attribute grouper. Maybe: 

grouper = getattr(obj, "grouper", False)

Reply:
Due to my misunderstanding of when save_model is executed please add the following logic to the file monkeypatch/models.py.

```
def save(func):
    """Override the FileAdmin save_model method"""

    def inner(self, *args, **kwargs):
        func(self, *args, **kwargs)

        grouper = getattr(self, "grouper", False)

        if self.id and grouper and not grouper.canonical_file_id:
            grouper.canonical_file_id = self.id
            grouper.save()

        return 
    return inner


filer.models.File.save = save(  # noqa: E305
    filer.models.File.save
)
```

Reply:
Updated